File: sample/CStyleSamplesVB/Samples/LatentSVM.vb
Imports System
Imports System.Collections.Generic
Imports System.Diagnostics
Imports OpenCvSharp

' Namespace OpenCvSharpSamplesVB
Imports SampleBase

''' <summary>
''' LatentSVM
''' </summary>
Friend Module LatentSVM
    Public Sub Start()
        Using detector As New CvLatentSvmDetector(FilePath.Text.LatentSvmCat)
            Using imageSrc As New IplImage(FilePath.Image.Cat, LoadMode.Color)
                Using imageDst As IplImage = imageSrc.Clone()
                    Using storage As New CvMemStorage()
                        Form1.TextBox1.AppendText(String.Format("Running LatentSVM..."))
                        Dim watch As Stopwatch = Stopwatch.StartNew()

                        Dim result As CvSeq(Of CvObjectDetection) = detector.DetectObjects(imageSrc, storage, 0.5F, 2)

                        watch.Stop()
                        Form1.TextBox1.AppendText(Environment.NewLine & String.Format("Elapsed time: {0}ms", watch.ElapsedMilliseconds))

                        For Each detection As CvObjectDetection In result
                            Dim boundingBox As CvRect = detection.Rect
                            imageDst.Rectangle(New CvPoint(boundingBox.X, boundingBox.Y), New CvPoint(boundingBox.X + boundingBox.Width, boundingBox.Y + boundingBox.Height), CvColor.Red, 3)
                        Next detection

                        Using TempCvWindow As CvWindow = New CvWindow("LatentSVM result", imageDst)
                            Cv.WaitKey()
                        End Using
                    End Using
                End Using
            End Using
        End Using
    End Sub
End Module
' End Namespace

File: sample/CStyleSamplesCS/Samples/LatentSVM.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Text;
using OpenCvSharp;
using SampleBase;

namespace CStyleSamplesCS
{
    /// <summary>
    /// LatentSVM
    /// </summary>
    class LatentSVM
    {
        public LatentSVM()
        {
            using (var detector = new CvLatentSvmDetector(FilePath.Text.LatentSvmCat))
            using (var imageSrc = new IplImage(FilePath.Image.Cat, LoadMode.Color))
            using (var imageDst = imageSrc.Clone())
            using (var storage = new CvMemStorage())
            {
                Console.WriteLine("Running LatentSVM...");
                Stopwatch watch = Stopwatch.StartNew();

                CvSeq<CvObjectDetection> result = detector.DetectObjects(imageSrc, storage, 0.5f, 2);

                watch.Stop();
                Console.WriteLine("Elapsed time: {0}ms", watch.ElapsedMilliseconds);

                foreach (CvObjectDetection detection in result)
                {
                    CvRect boundingBox = detection.Rect;
                    imageDst.Rectangle(
                        new CvPoint(boundingBox.X, boundingBox.Y), 
                        new CvPoint(boundingBox.X + boundingBox.Width, boundingBox.Y + boundingBox.Height),
                        CvColor.Red, 3);
                }

                using (new CvWindow("LatentSVM result", imageDst))
                {
                    Cv.WaitKey();
                }
            }
        }
    }
}